src/SignalR/common/Protocols.Json/src/JsonHubProtocolOptions.cs

```diff
@@ -21,5 +21,20 @@ public JsonHubProtocolOptions()
         public bool IgnoreNullValues { get => _serializerOptions.IgnoreNullValues; set => _serializerOptions.IgnoreNullValues = value; }
         public bool WriteIndented { get => _serializerOptions.WriteIndented; set => _serializerOptions.WriteIndented = value; }
         public bool AllowTrailingCommas { get => _serializerOptions.AllowTrailingCommas; set => _serializerOptions.AllowTrailingCommas = value; }
+        public bool UseCamelCase
```

Comment:
Do we not want to expose our own enum rather than mapping a bool to an enum? I guess this can just be part of the api review.

Reply:
`JsonNamingPolicy` is an abstract base class. `JsonNamingPolicy.CamelCase` is a static instance of `CamelCaseNamingPolicy`.

The question is whether it makes sense for people to configure their own naming policy. If so then expose PropertyNamingPolicy. If someone wants camelcase then they can set `JsonNamingPolicy.CamelCase`. If they want something else then they can assign their own implementation of JsonNamingPolicy.

Reply:
Get consensus here before merging please :)

Reply:
We're changing this in preview 6 and have filed https://github.com/aspnet/AspNetCore/issues/9519 to review the API

Reply:
There are preview 6 changes planned for JSON that will likely make this entirely moot. We may be able to just expose their options directly in that. We can't use their types right now because of the sources package.

Reply:
Followed up with @JamesNK offline. I'm OK with this as the preview 5 version of the API given the review in preview 6.